src/main/java/org/folio/service/email/EmailService.java

```diff
@@ -50,33 +49,30 @@ public class EmailService {
 
   private final RestClient configurationRestClient;
   private final RestClient userRestClient;
-  private final DBClientFactory dbClientFactory;
   private final LedgerDAO ledgerDAO;
 
-  public EmailService(RestClient configurationRestClient, RestClient userRestClient, DBClientFactory dbClientFactory, LedgerDAO ledgerDAO) {
+  public EmailService(RestClient configurationRestClient, RestClient userRestClient, LedgerDAO ledgerDAO) {
     this.configurationRestClient = configurationRestClient;
     this.userRestClient = userRestClient;
-    this.dbClientFactory = dbClientFactory;
     this.ledgerDAO = ledgerDAO;
   }
 
-  public void createAndSendEmail(RequestContext requestContext, LedgerFiscalYearRollover rollover) {
+  public void createAndSendEmail(RequestContext requestContext, LedgerFiscalYearRollover rollover, DBConn conn) {
     logger.debug("createAndSendEmail:: Trying to create and send email");
     getHostAddress(requestContext)
       .onSuccess(hostAddressResponse -> getCurrentUser(requestContext)
-        .onSuccess(userResponse -> {
-          DBClient client = dbClientFactory.getDbClient(requestContext);
-          ledgerDAO.getLedgerById(rollover.getLedgerId(), client)
-            .onSuccess(ledger -> {
-              String hostAddress = hostAddressResponse.getJsonArray(CONFIGS_KEY).getJsonObject(0).getMap().get(VALUE_KEY).toString();
-              String linkToRolloverLedger = createRolloverLedgerLink(hostAddress, rollover.getLedgerId());
-              Map<String, String> headers = getHeaders(requestContext);
-              EmailEntity emailEntity = getEmailEntity(rollover, linkToRolloverLedger, ledger.getName(), userResponse);
-              logger.info("createAndSendEmail:: Sending email");
-
-              sendEmail(requestContext, headers, emailEntity);
-            }).onFailure(t -> logger.error("createAndSendEmail:: Getting ledger failed {}", t.getMessage()));
-        }).onFailure(t -> logger.error("createAndSendEmail:: Getting user failed {}", t.getMessage())))
+        .onSuccess(userResponse -> ledgerDAO.getLedgerById(rollover.getLedgerId(), conn)
+          .onSuccess(ledger -> {
```

Comment:
**Question**
Can we use `compose` instead of nested `onSuccess` chains? Log messages are more or less similar and adding a stack trace should make it clear which `compose` failed.